### ISSUE TITLE
Help getting access to custom tile data

### ISSUE THREAD

**Issue by ninjatronic** [Cammin/LDtkUnity#39]
Hi. Apologies for the noob question but I'm coming up against a brick wall here.

I have a tileset "Ground" which has custom data for each tile using the following schema...

```
{
  "movementSpeed": 0.8
}
```

I've imported the world into unity and put the level in the scene. In code I can get the tile under the player's current position...
```
public class PlayerController : MonoBehaviour
{

    Rigidbody2D body;
    Tilemap ground;
    private float horizontal, vertical;
    private readonly float runSpeed = 1.0f, moveLimiter = 0.7f;

    void Start()
    {
        body = GetComponent<Rigidbody2D>();
        ground = GameObject.Find("Ground").GetComponentInChildren<Tilemap>();
    }

    private void FixedUpdate()
    {
        Vector3Int position = Vector3Int.RoundToInt(body.position);
        TileBase tile = ground.GetTile(position);
    }
}
```

But from here I am drawing a blank on how to access the custom data associated with this tile. Any help is much appreciated. Thanks for the excellent project so far it has been very useful.

**Comment by Cammin** [comment]
Hi, there is indeed a way to get the custom data defined from the importer. I have plans to make this experience easier, but for now, you can access the tile data by deserializing the json data, then traversing through it until you can access the CustomData from the TilesetDefinition:  
![image](https://user-images.githubusercontent.com/55564581/181204518-929e7895-9b0d-47a5-8ba0-be253af7233f.png)

You can access and deserialize the data from this scriptableobject asset from the imported project with the function ``FromJson()``:
![image](https://user-images.githubusercontent.com/55564581/181204983-5a6f2b91-e62a-4e0c-b1fe-3bcd95850f95.png)

Let me know if this solution can work for you 🙂

**Comment by ninjatronic** [comment]
My LDTK project saves levels as separate files and as such I'm struggling to find a way to access this JSON object in code. I can see it in the assets in the project explorer but when I drop the LDTK project file into the scene in the hierarchy it does not show these children. Is it possible for you to show a simple code example as to how I would access this? Sorry for the noob question, I'm new to Unity and LDTK. Many thanks.

**Comment by Cammin** [comment]
This [function](https://cammin.github.io/LDtkToUnity/api/LDtkUnity.Editor.LDtkPostprocessor.html#LDtkUnity_Editor_LDtkPostprocessor_OnPostprocessLevel_UnityEngine_GameObject_LDtkUnity_LdtkJson_) in particular will be useful for your situation, where you can traverse the project json to access definitions and additionally levels and their used tilesets in that way.

And you can check [here](https://cammin.github.io/LDtkToUnity/documentation/Topics/topic_CustomImporting.html#ldtkpostprocessor) to understand how the LDtk postprocessing works.

Also feel free to join the [Discord](https://discord.gg/7RPGAW9dJx) server where I can provide quicker answers. I didn't realize you had responded here recently after my previous post 😅

**Comment by Cammin** [comment]
Another significantly easier option is defining intgrid values to describe where your character should change it's speed. then you can make your character check for an IntGridTile existing at a position, and check if it matches a reference you are checking for. And voila :)
Let me know if that can work out for you.

**Comment by Cammin** [comment]
Custom tile data is now available in the newest update! (4.0.0) 
You get a tileset tile from the tilemap, which will contain both the custom data and enum values.
![image](https://github.com/Cammin/LDtkToUnity/assets/55564581/79cc5a3d-319c-49fc-879f-28959d7cbb5c)
